xikolo-ios/Helpers/AnalyticsHelper.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  AnalyticsHelper.swift
+//  xikolo-ios
+//
+//  Created by Bjarne Sievers on 27.08.17.
+//  Copyright © 2017 HPI. All rights reserved.
+//
+
+import Foundation
+
+class AnalyticsHelper {
+
+    class func setup() {
+        NotificationCenter.default.addObserver(self, selector: #selector(AnalyticsHelper.check), name: NotificationKeys.reachabilityChanged, object: nil)
+        check()
+    }
+
+    @objc class func check() {
+        do {
+            let eventCount = try CoreDataHelper.backgroundContext.count(for: TrackingHelper.getTrackingEventRequest())
+            if eventCount > 0 {
+                self.uploadTrackingEvents()
+            }
+        } catch {
+            print("\(error)")
+        }
+    }
+
+    class func uploadTrackingEvents() {
+        guard ReachabilityHelper.reachability.currentReachabilityStatus == .reachableViaWiFi else { return }
+        var events: [TrackingEvent]
+        do {
+            events = try CoreDataHelper.backgroundContext.fetch(TrackingHelper.getTrackingEventRequest())
+            guard events.count > 0 else { return }
+            for event in events {
+                SpineHelper.save(TrackingEventSpine(event)).onSuccess(callback: { (_) in
+                    CoreDataHelper.backgroundContext.delete(event)
+                })
```

Comment:
In my opinion, here we could run into some timing issues. I would suggest a more sophisticated approach.
```swift
events.traverse { event in
  return SpineHelper.save(TrackingEventSpine(event)).map { _ -> Future<Void, NoError>
    CoreDataHelper.backgroundContext.delete(event)
    return Future { complete in complete(.success()) }
  }
}.onComplete { _ in
  CoreDataHelper.saveContext()  // important!
  // all 10 requests have been processed (network + core data)
  // no need to wait 60 secs
  // call check()
}
```